cfgmgr/vxlanmgr.cpp

```diff
@@ -585,10 +911,125 @@ bool VxlanMgr::deleteVxlan(const VxlanInfo & info)
+                                   std::string src_ip, std::string dst_ip, std::string vlan_id)  
+{
+    int ret = 0;
+
+    //ip link add <vxlan_dev_name> type vxlan id <vni> local <src_ip> remote <dst_ip> dstport 4789
+    //ip link set <vxlan_dev_name> master DOT1Q_BRIDGE_NAME
+    //bridge vlan add vid <vlan_id> dev <vxlan_dev_name>
+    //ip link set <vxlan_dev_name> up
+    std::string vxlan_dev_name;
+    vxlan_dev_name = std::string("") + std::string(vxlanTunnelName) + "-" + std::string(vlan_id);
+
+    SWSS_LOG_NOTICE("Kernel tnl_name: %s vni_id: %s src_ip: %s dst_ip:%s vlan_id: %s",
+                vxlanTunnelName.c_str(), vni_id.c_str(), src_ip.c_str(), dst_ip.c_str(), vlan_id.c_str());
+    /* Case 1: Entry exist - Erase from cache & return
+     * Case 2: Enry does not exist - Create netDevice in Kernel
+     * Case 3: Entry exist but modified - Not taken care. Will address later
+     */
+    if (true == m_in_reconcile)
+    { 
+        auto it = m_vxlanNetDevices.find(vxlan_dev_name);
+        if (it != m_vxlanNetDevices.end())
+        {
+            m_vxlanNetDevices.erase(it);
+            SWSS_LOG_NOTICE("Reconcile VxlanNetDevice %s reconciled. Pending %lu",
+                            vxlan_dev_name.c_str(), m_vxlanNetDevices.size());
+            return 0;
+        }
+        else
+        {
+            SWSS_LOG_NOTICE("Reconcile VxlanNetDevice %s doesn't not exist. Pending %lu",
+                            vxlan_dev_name.c_str(), m_vxlanNetDevices.size());
+        }
+    }
+    else
+    {
+        SWSS_LOG_NOTICE("Creating VxlanNetDevice %s", vxlan_dev_name.c_str());
+    }
+
+    std::string res, chkcmd, cmds;
+    chkcmd = "ip link show " + vxlan_dev_name;
+    ret = swss::exec(chkcmd,res);
+    if(ret == 0)
+    {
+        cmds = "ip link del dev " + vxlan_dev_name ;
+        EXEC_WITH_ERROR_THROW(cmds, res);
+    }
+    cmds = std::string("")
+        + BASH_CMD + " -c \""
+        + IP_CMD + " link add " + vxlan_dev_name + " type vxlan id " + std::string(vni_id) + " address " + gMacAddress.to_string()
+        + " local " + src_ip + ((dst_ip  == "")? "":(" remote " + dst_ip)) + " nolearning " +" dstport 4789 " + " && "
+        + IP_CMD + " link set " + vxlan_dev_name + " master Bridge " 
+        + " && "
+        + BRIDGE_CMD + " vlan add vid " + std::string(vlan_id) + " dev " + vxlan_dev_name 
+        + " && "
+        + BRIDGE_CMD + " vlan add vid " + std::string(vlan_id) + " untagged pvid dev " + vxlan_dev_name 
+        + " && ";
+    if( vlan_id != "1")
+    {
+        cmds = cmds + BRIDGE_CMD + " vlan del vid 1 dev " + vxlan_dev_name + " && ";
+    }
+    cmds = cmds + IP_CMD + " link set " + vxlan_dev_name + " up " + "\"";
+    EXEC_WITH_ERROR_THROW(cmds, res);
+    return ret;
+}
+
+int VxlanMgr::deleteVxlanNetdevice(std::string vxlan_dev_name)
+{    
+    int ret = 0;
+    std::string res;
+    const std::string cmd = std::string("") + IP_CMD " link del dev " + vxlan_dev_name;
```

Comment:
miss `+` between `IP_CMD` and `" link del dev "`

Reply:
done